src/tile_id.cpp

```diff
@@ -19,10 +19,20 @@ limitations under the License. */
 
 std::string tileURL(TileId const& tile_id)
 {
-  auto url = tile_id.tile_server;
+  auto url = tile_id.tile_server;  
+
+  // compute latitude and longitude from tile coordinates
+  float n = pow(2, tile_id.zoom);
+  float lon_deg = float(tile_id.coord.x) / float(n) * 360.0f - 180.0f;
+  float lat_deg = 180.0f / M_PI * atan(sinh(M_PI * (1 - 2 * tile_id.coord.y / n)));
+
+  // substitute placeholders
+  boost::replace_all(url, "{lat}", std::to_string(lat_deg));
+  boost::replace_all(url, "{lon}", std::to_string(lon_deg));
```

Comment:
There already is a dedicated function to convert tile ID to WGS coords.

Also, std::to_string fails on machines with a locale that renders floats with a decimal comma, e.g. German or Czech. To get locale-independent stringified version of the float, you have to use either boost::lexical_cast, or std::to_chars (available only in C++17 and newer).

```suggestion
  // compute latitude and longitude from tile coordinates
  const auto wgs = toWGSCoordinate(tile_id.coord, tile_id.zoom);

  // substitute placeholders
  boost::replace_all(url, "{lat}", boost::lexical_cast<std::string>(wgs.lat));
  boost::replace_all(url, "{lon}", boost::lexical_cast<std::string>(wgs.lon));
```